Option to check if owner is in a server and leave if they're not.

AleanaA opened the pull request:
After creating your pull request, tick these boxes if they are applicable to you.

- [x] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.5/3.6

----

### Description
Implements an option to check if the owner shares the servers with the bot. If the owner doesn't share the server with the bot, the bot will leave that server.

Credit to Winter for the idea and most of the code o//

### Related issues (if applicable)

AleanaA committed:
Implement Winter's code and make it work with rewrite

jayktaylor commented on musicbot/bot.py:
```diff
@@ -939,6 +939,11 @@ def run(self):
                 ser = ('{} (unavailable)'.format(s.name) if s.unavailable else s.name)
                 log.info(' - ' + ser)
+                if self.config.leavenonowners:
+                    check = s.get_member(owner.id)
```
This may have an unintended result if the server is unavailable.

AutumnClove replied:
I'll add in a try and except?

jayktaylor replied:
I have a suspicion that if a server is unavailable and you try to get a member, it'll return None regardless of if the member is in there or not because the data may not be obtainable. I'd rather err on the side of caution here and check to make sure that the server is available first.

AutumnClove replied:
Makes sense, I'll add in
```
If s.unavailable:
    unavailable_servers += 1
    log.info('Not going through with a check in {} servers due to unavailability.'format(unavailable_servers))
```
As well as a counter right below `if self.config.leavenonowners`

jayktaylor commented on config/example_options.ini:
```diff
@@ -144,6 +144,9 @@ ShowConfigOnLaunch = no
 LegacySkip = no
 
+# Leave servers if the owner is not found in them.
+LeaveNonOwners = no
```
Maybe we can come up with a more descriptive name for this config option?

AutumnClove replied:
I could understand that I just didn't want to make it to long, and keep it the same as the other config options. Will think of a better name.

AutumnClove replied:
How does LeaveNonBotOwners sound to you?

BabyBoySnow committed:
Unavailavilty check, rename config.

AleanaA committed:
this should make the check work here.

AleanaA committed:
Make options more descriptive and fix config

AleanaA committed:
If we put that there, it'll repeat for every unavailable server.

BabyBoySnow committed:
More descriptive, message owner

Adds a little more description as to why we left, also messages the owner when we leave.

AleanaA committed:
Quick fix for unavailable servers, and *rewrite*

jayktaylor reviewed (changes requested):
Can you support strings from i18n files for the messages you send to the owner of the bot?

BabyBoySnow committed:
Added i18n support

AutumnClove committed:
Merge branch 'review' into patch-1
